src/Fixer/ClassNotation/ConstTypeRequiredFixer.php

```diff
@@ -0,0 +1,385 @@
+            $type = 'int';
+        }
+
+        $this->insertTypeForConst($tokens, $constIndex + 1, $type ?? 'mixed');
+    }
+
+    private function fixSimpleStringConstant(
+        Tokens $tokens,
+        int $constIndex,
+        int $valueIndex
+    ): void {
+        $type = $this->getTypeForStringToken($tokens[$valueIndex]);
+        $this->insertTypeForConst($tokens, $constIndex + 1, $type ?? 'mixed');
+    }
+
+    private function fixArrayConstant(
+        Tokens $tokens,
+        int $constIndex,
+        int $valueIndex
+    ): void {
+        if ($tokens[$valueIndex]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $valueIndex);
+        } else {
+            \assert($tokens[$valueIndex]->isGivenKind(T_ARRAY));
+            $endIndex = $tokens->getNextMeaningfulToken($valueIndex);
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $endIndex);
+        }
+
+        $endIndex = $tokens->getNextMeaningfulToken($endIndex);
+        $this->insertTypeForConst($tokens, $constIndex + 1, $tokens[$endIndex]->equals(';') ? 'array' : 'mixed');
+    }
+
+    private function insertTypeForConst(Tokens $tokens, int $index, string $type): void
+    {
+        $tokens->insertSlices(
+            [
+                $index => [
+                    new Token([T_WHITESPACE, ' ']),
+                    new Token(['array' === $type ? CT::T_ARRAY_TYPEHINT : T_STRING, $type]),
+                ],
+            ],
+        );
+    }
+
+    private function getTypeForStringToken(Token $token): ?string
+    {
+        \assert($token->isGivenKind(T_STRING));
+
+        $content = $token->getContent();
+        $contentLower = strtolower($content);
+
+        if ('true' === $contentLower || 'false' === $contentLower) {
+            $type = 'bool';
+        } elseif ('null' === $contentLower) {
+            $type = 'null';
+        } elseif ($this->isPredefinedConstStr($content)) {
+            $type = 'string';
+        } elseif ($this->isPredefinedConstInt($content)) {
+            $type = 'int';
+        } elseif ($this->isPredefinedConstFloat($content)) {
+            $type = 'float';
+        } else {
+            $type = null;
+        }
+
+        return $type;
+    }
+
+    private function isMagicConstOfTypeInt(Token $token): bool
+    {
+        $magicConstInt = [
+            T_LINE,
+        ];
+
+        return $token->isGivenKind($magicConstInt);
+    }
+
+    private function isMagicConstOfTypeStr(Token $token): bool
+    {
+        $magicConstStr = [
+            T_FILE,
+            T_DIR,
+            T_CLASS_C,
+            T_NS_C,
+            T_TRAIT_C,
+        ];
+
+        return $token->isGivenKind($magicConstStr);
+    }
+
+    private function isPredefinedConstStr(string $content): bool
+    {
+        static $predefinedConstStr = null;
+
+        if (null === $predefinedConstStr) {
```

Comment:
any need for this `if` ?

Reply:
to only create the list:
- when needed @ runtime
- create only max one time